PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -338,21 +338,27 @@ def miniAOD_customizeCommon(process):
     for idmod in photon_ids:
         setupAllVIDIdsInModule(process,idmod,setupVIDPhotonSelection,None,False,task)
 
-    #---------------------------------------------------------------------------
-    #Adding  Boosted Subjets taus
+    #-- Adding boosted taus
     from RecoTauTag.Configuration.boostedHPSPFTaus_cfi import addBoostedTaus
     addBoostedTaus(process)
-    #---------------------------------------------------------------------------
-    #Adding tau reco for 80X legacy reMiniAOD
-    #make a copy of makePatTauTask to avoid labels and substitution problems
-    _makePatTausTaskWithTauReReco = process.makePatTausTask.copy()
-    #add PFTau reco modules to cloned makePatTauTask
     process.load("RecoTauTag.Configuration.RecoPFTauTag_cff")
-    _makePatTausTaskWithTauReReco.add(process.PFTauTask)
-    #replace original task by extended one for the miniAOD_80XLegacy era
+    process.load("RecoTauTag.Configuration.HPSPFTaus_cff")
+    #-- Adding customization for 94X 2017 legacy reMniAOD
+    from Configuration.Eras.Modifier_run2_miniAOD_94XFall17_cff import run2_miniAOD_94XFall17
+    _makePatTausTaskWithRetrainedMVATauID = process.makePatTausTask.copy()
+    _makePatTausTaskWithRetrainedMVATauID.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask)
+    run2_miniAOD_94XFall17.toReplaceWith(
+        process.makePatTausTask, _makePatTausTaskWithRetrainedMVATauID
+        )
+    #-- Adding customization for 80X 2016 legacy reMiniAOD
     from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+    _makePatTausTaskWithTauReReco = process.makePatTausTask.copy()
+    _makePatTausTaskWithTauReReco.add(process.PFTauTask)
+    _makePatTausTaskWithTauReReco.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask)
```

Comment:
While looking at the list of running modules, I see that there is no change in run2_miniAOD_80XLegacy setup.

isn't this addition of hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask actually redundant
because it's already a part of process.PFTauTask added just one line above?

If the answer is yes, please remove L357

Reply:
Hi Slava, 
thank you for the careful review. Indeed the module hpsPFTauDiscriminationByVLooseIsolationMVArun2v1DBoldDMwLT needs to be added to the Task and should make a change. But I've removed the hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask (w/o hpsPFTauDiscriminationByVLooseIsolationMVArun2v1DBoldDMwLT), which indeed is already part of PFTauTask. And I have removed the redundant initialization of the module.

Cheers,
Roger

Reply:
Hi Roger,

So, just to be explicit: PFTauTask includes produceAndDiscriminateHPSPFTausTask, which includes hpsPFTauMVAIsolation2Task, which in turn includes hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask.
This means that 
```python
_makePatTausTaskWithTauReReco.add(process.PFTauTask)
_makePatTausTaskWithTauReReco.add(process.hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask)
```
is exactly the same in its final result as 
```python
_makePatTausTaskWithTauReReco.add(process.PFTauTask)
```


Do I understand your last response correctly that you are going to edit this file (PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py ) in this PR to remove the redundant line adding hpsPFTauDiscriminationByIsolationMVArun2v1DBoldDMwLTTask to _makePatTausTaskWithTauReReco?
Please confirm

Thank you.

Slava

Reply:
Confirmed and uploaded...